### DISCUSSION TITLE
How can I pass in parameters from one to another ViewModel for its View to use?

### DISCUSSION THREAD

**Discussion by Wind-Explorer** [AvaloniaUI/Avalonia#9981] [Q&A] [answered]
I have an app that should display data based on variables from parameters received from calls of the ViewModel.

I have noticed two places in which the ViewModel gets called from, one is the intended call by the parent ViewModel for displaying the View of the child ViewModel with the help of `ViewLocator.cs`. Another is the `DataContext` required by the View to enable data binding.

Former: \[`ParentViewModel.cs`\]

```cs
public ParentViewModel()
{
    UserControlContent = new ChildViewModel(genericParameter: "actual parameter");
}
```

Latter: \[`ChildView.axaml.cs`\]

```cs
public ChildView()
{
    InitializeComponent();
    DataContext = new ChildViewModel(genericParameter: "not the parameter I want");
}
```

So I want to pass in the `"actual parameter"` as shown above for show in the resulting View. I expect the final view to look like this:

[![image of an app window showing the text "actual parameter".](https://i.stack.imgur.com/aTlUW.png)](https://i.stack.imgur.com/aTlUW.png)

But in reality I get this:

[![image of an app window showing the text "not the parameter I want".](https://i.stack.imgur.com/E2OPY.png)](https://i.stack.imgur.com/E2OPY.png)

So, how can I work around this and get the View to display the right data?

Appreciate any input!

**Comment by rabbitism** [comment, marked as answer]
You don't need to assign child view model here to enable binding.

**Reply by Wind-Explorer** [reply]
In that case, would you mind demonstrating the proper way to enable data binding?

**Reply by rabbitism** [reply]
You can check docs here, https://docs.avaloniaui.net/docs/getting-started/ide-support
Design.DataContext part.

**Reply by Wind-Explorer** [reply]
> You can check docs here, https://docs.avaloniaui.net/docs/getting-started/ide-support
> Design.DataContext part.

Thank you for the hint! Back when before my code got to this level of complexity I needed `DataContext` for my View to display sample data, but now I thought I'd remove `DataContext` and see what happens, and that turns out to resolve the issue. 🫠
